### DISCUSSION TITLE
Change tab position programmatically & get current tab position

### DISCUSSION THREAD

**Discussion by Softkeydel** [react-native-community/react-native-tab-view#1285] [Q&A]
Current behavior

`
const renderScene = ({ route, jumpTo }) => { switch (route.key) { case 'music': return <MusicRoute jumpTo={jumpTo} />; case 'albums': return <AlbumsRoute jumpTo={jumpTo} />; } };
`

Here jumpTo() is accessible to the child components.

Expected behavior
Change tab position programmatically & get current tab position

I need to change the tab position on a button click on the same component the tab view is residing & aslo get the current tab position.

Please help

**Comment by satya164** [comment]
You have the tab position in your state. To access current position, you use your `index` in state, and to programmatically change it, you update your index state.

**Reply by Softkeydel** [reply]
@satya164 
Yes I've the index but if I change the index with state it's not changing the tab.

Could you please help me

**Reply by satya164** [reply]
Show your code

**Reply by Softkeydel** [reply]
I'll create a demo file & update you

**Reply by Softkeydel** [reply]
@satya164 

I was doing some mistakes.
Now it's working.

Thanks
